server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java

```diff
@@ -47,11 +49,14 @@
      */
     public static BoolQueryBuilder buildQuery(Criteria criteria, String timeProperty,
             String businessTxnProperty, Class<?> targetClass) {
-        long startTime = criteria.calculateStartTime();
-        long endTime = criteria.calculateEndTime();
-
+        /**
+         * Internally all time units are stored in microseconds
+         * Criteria API accepts milliseconds, therefore range adjustment is needed
+         */
         BoolQueryBuilder query = QueryBuilders.boolQuery()
-                .must(QueryBuilders.rangeQuery(timeProperty).from(startTime).to(endTime));
+                .must(QueryBuilders.rangeQuery(timeProperty)
+                        .from(TimeUnit.MILLISECONDS.toMicros(criteria.calculateStartTime()))
```

Comment:
Ah yes, I was probably getting too caught up with the timestamp mapping issue. So might not be a problem after all :)